### PR TITLE
Add new SignPostIntegration instrumentation

### LINKED ISSUE #179: Integrate with os.signpost functionality
It would be great integrating Opentelemetry with system signpost functionality. It would allow reusing all the opentelemetry instrumentation when doing profiling for example.
Using a dedicated SpanProcessor that simply calls `os_signpost_interval_begin` and `os_signpost_interval_end` when starting or ending the span would be great.

### PR DESCRIPTION
**Pull request by nachoBonafonte**
It integrateds spans created in opentelemetry with os_signpost functionality, allowing the spans to appear automatically in `Instruments` when profiling the app:

Screenshot 2021-05-20 at 15 10 33

Closes #179

### CODE REVIEW

File: Sources/Instrumentation/SignPostIntegration/SignPostIntegration.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2020, OpenTelemetry Authors
```
Comment (SergeyKanzhelev) [inline Sources/Instrumentation/SignPostIntegration/SignPostIntegration.swift:1]: let's remove the year everywhere =)

File: Sources/Instrumentation/SignPostIntegration/SignPostIntegration.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+@available(macOS 10.14, *)
+public class SignPostIntegration: SpanProcessor {
+    public let isStartRequired = true
+    public let isEndRequired = true
+    public let osLog = OSLog(subsystem: "OpenTelemetry", category: .pointsOfInterest)
+
+    public init() {}
+
+    public func onStart(parentContext: SpanContext?, span: ReadableSpan) {
+        let signpostID = OSSignpostID(log: osLog, object: self)
```
Comment (SergeyKanzhelev) [inline Sources/Instrumentation/SignPostIntegration/SignPostIntegration.swift:32]: is this static data that can be added to the resource?

File: Sources/Instrumentation/SignPostIntegration/SignPostIntegration.swift (hunk 1 above)
Comment (nachoBonafonte) [reply Sources/Instrumentation/SignPostIntegration/SignPostIntegration.swift:1]: I will replace with the format I found in opentelemetry-java:
```
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */
```
If approved I will create another PR changing to this header in all project files

File: Sources/Instrumentation/SignPostIntegration/SignPostIntegration.swift (hunk 2 above)
Comment (nachoBonafonte) [reply Sources/Instrumentation/SignPostIntegration/SignPostIntegration.swift:32]: No, it creates a different ID each time it is called

**Review by SergeyKanzhelev — APPROVED** [review]

**Pull request merged**